### PR TITLE
Merge Staging and master

### PR DESCRIPTION
**Pull request by MemunaHaruna**
#### What does this PR do?
* This PR is to merge staging and master branch.

#### Description of Task to be completed?
* N/A

#### How should this be manually tested?
* N/A

#### Any background context you want to provide?
* N/A

#### What are the relevant pivotal tracker stories?
* N/A

#### Screenshots (if appropriate)
* N/A

#### Questions
* N/A

### CODE REVIEW

File: client/app/actionTypes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// document action-types
+export const CREATE_DOCUMENT = 'CREATE_DOCUMENT';
```
Comment (femisenjobi) [inline client/app/actionTypes.js:2]: What if you have this?
```
export const DOCUMENTS = {
   VIEW: 'VIEW_DOCUMENTS',
   EDIT: 'EDIT_DOCUMENT',
   DELETE: 'DELETE_DOCUMENT',
   VIEW_ERROR: 'VIEW_DOCUMENTS_ERROR',
 ...
```
You can import with:
```
   import {DOCUMENTS} from '../actionTypes'
```
And use as
```DOCUMENTS.CREATE```
It's nothing serious though just neater. If you got more ti.me after the other reviews, you can do that

File: client/app/actions/documentActions.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import toastr from 'toastr';
+import { VIEW_DOCUMENTS, VIEW_DOCUMENTS_ERROR, CREATE_DOCUMENT,
+  CREATE_DOCUMENT_ERROR, CHANGE_CURRENT_DOCUMENT, EDIT_DOCUMENT,
+  EDIT_DOCUMENT_ERROR, DELETE_DOCUMENT, DELETE_DOCUMENT_ERROR,
+  SEARCH_DOCUMENT, SEARCH_DOCUMENT_ERROR, VIEW_ONE_DOCUMENT,
+  VIEW_ONE_DOCUMENT_ERROR } from '../actionTypes';
+
+
+export const viewAllDocuments = (url) => {
+  console.log(url, 'urls');
```
Comment (femisenjobi) [inline client/app/actions/documentActions.js]: You can leave the console logs out of your localhost

File: client/tests/components/SearchDocuments.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  });
+  afterEach(() => {
+    props.searchForDocuments.reset();
+  });
+
+  it('should find the neccessary dom elements ', () => {
+    expect(component.find('.search-docs').length).to.equal(1);
+    expect(component.find('input').length).to.equal(1);
+  });
+  it('simulates click events', () => {
```
Comment (femisenjobi) [inline client/tests/components/SearchDocuments.spec.js:24]: Your test description should be what is asserted not what you did. `calls searchForDocuments once` sounds more like it. You update that everywhere else.

File: client/tests/components/Profile.spec.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  it('should have all the set props', () => {
+    expect(component.props().user.id).to.equal(7);
+  });
+  it('should find the neccessary dom elements ', () => {
+    component.setState({ isEdit: true, isChecked: true });
+    expect(component.find('.switch').length).to.equal(1);
+    expect(component.find('input').length).to.equal(7);
+  });
+  xit('simulates click events', () => {
```
Comment (femisenjobi) [inline client/tests/components/Profile.spec.js]: Why do you wanna skip this?

File: client/tests/components/DocumentCard.spec.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { shallow, mount } from 'enzyme';
+import { DocumentCard } from '../../app/components/documents/DocumentCard.jsx';
+
+describe('DocumentCard component', () => {
+  let component;
+  const props = {
+    currentDocument: {},
+    documents: [{ id: 1, title: '1st document', content: 'lorem ipsum', userId: 3, createdAt: '12-06-2017', user: { username: 'toodoo' } },
```
Comment (femisenjobi) [inline client/tests/components/DocumentCard.spec.js]: Since you have fakeData you can clean this up by transferring documents there

**Review by femisenjobi — COMMENTED** [review]
Ater a quick glance, I must say I love what I see. Clean work and thorough tests. I dropped a few comments to further clean up your work. I'll spend some more time looking later

File: client/app/actionTypes.js (hunk 1 above)
Comment (MemunaHaruna) [reply client/app/actionTypes.js:2]: Alright. Noted

**Pull request closed without merging**